front/lib/swr/data_source_view_documents.ts

```diff
@@ -0,0 +1,205 @@
+} from "@dust-tt/types";
+import assert from "assert";
+import type { Fetcher } from "swr";
+import type { SWRMutationConfiguration } from "swr/mutation";
+import useSWRMutation from "swr/mutation";
+
+import { useDataSourceViewContentNodes } from "@app/lib/swr/data_source_views";
+import { fetcher, fetcherWithBody, useSWRWithDefaults } from "@app/lib/swr/swr";
+import type { GetDataSourceViewDocumentResponseBody } from "@app/pages/api/w/[wId]/spaces/[spaceId]/data_source_views/[dsvId]/documents/[documentId]";
+import type { PostDocumentResponseBody } from "@app/pages/api/w/[wId]/spaces/[spaceId]/data_sources/[dsId]/documents";
+import type { PatchDocumentResponseBody } from "@app/pages/api/w/[wId]/spaces/[spaceId]/data_sources/[dsId]/documents/[documentId]";
+
+// Centralized way to get urls -> reduces key-related inconcistencies
+function getUrlHasValidParameters(
+  method: "POST" | "GET" | "PATCH",
+  documentId?: string
+): documentId is string {
+  // Only require documentId for GET and PATCH methods
+  return method === "POST" || !!documentId;
+}
+const getUrl = ({
+  method,
+  owner,
+  dataSourceView,
+  documentId,
+}: {
+  method: "POST" | "GET" | "PATCH";
+  owner: LightWorkspaceType;
+  dataSourceView: DataSourceViewType;
+  documentId?: string;
+}) => {
+  assert(
+    getUrlHasValidParameters(method, documentId),
+    "Cannot get or patch a document without a documentId"
+  );
+
+  const baseUrl = `/api/w/${owner.sId}/spaces/${dataSourceView.spaceId}`;
+  switch (method) {
+    case "POST":
+      return `${baseUrl}/data_sources/${dataSourceView.dataSource.sId}/documents`;
+    case "PATCH":
+      return `${baseUrl}/data_sources/${dataSourceView.dataSource.sId}/documents/${encodeURIComponent(documentId)}`;
+    case "GET":
+      return `${baseUrl}/data_source_views/${dataSourceView.sId}/documents/${encodeURIComponent(documentId)}`;
+  }
+};
+
+export function useDataSourceViewDocument({
+  dataSourceView,
+  documentId,
+  owner,
+  disabled,
+}: {
+  dataSourceView: DataSourceViewType | null;
+  documentId: string | null;
+  owner: LightWorkspaceType;
+  disabled?: boolean;
+}) {
+  const dataSourceViewDocumentFetcher: Fetcher<GetDataSourceViewDocumentResponseBody> =
+    fetcher;
+  const url =
+    dataSourceView && documentId
+      ? getUrl({
+          method: "GET",
+          owner,
+          dataSourceView,
+          documentId,
+        })
+      : null;
+
+  const { data, error, mutate } = useSWRWithDefaults(
+    url,
+    dataSourceViewDocumentFetcher,
+    {
+      disabled,
+    }
+  );
+
+  return {
+    document: data?.document,
+    isDocumentLoading: !disabled && !error && !data,
+    isDocumentError: error,
+    mutateDocument: mutate,
+  };
+}
+
+async function sendPatchRequest(
+  url: string,
+  {
+    arg,
+  }: {
+    arg: {
+      documentBody: PatchDataSourceWithNameDocumentRequestBody;
+    };
+  }
+) {
+  const res = await fetcherWithBody([url, arg.documentBody, "PATCH"]);
```

Comment:
There is one last thing, using `SWRMutation` bypasses the commit header logic we have (to invalidate some commits). It can be tackle in a follow up PR, but let's make sure to still add the `X-Commit-Hash`:

https://github.com/dust-tt/dust/blob/fc2e1e5dc516ebf220e9d3f185d9c73747922d3c/front/lib/swr/swr.ts#L111-L114